### PR TITLE
Use gulp for building

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -23,6 +23,15 @@ If you've written code that fixes an issue, `create a pull request
 <https://help.github.com/articles/creating-a-pull-request/>`_ (PR) so it's easy
 for us to incorporate your changes.
 
+Setting up for Development
+################
+
+We recommend using a `virtualenv <https://virtualenv.pypa.io/en/latest/>`_ to
+install the required packages in ``requirements.txt``. In addition, we use
+`gulp <http://gulpjs.com/>`_ as a frontend build system.  To build the frontend
+resources you can run ``make build dist`` once `npm <https://www.npmjs.com/>`_
```
Comment (derek-schultz) [inline docs/source/contributing.rst]: `build_dist`

File: docs_requirements.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Packages needed to build docs on https://readthedocs.org
-Django==1.8.3
+Django==1.9.4
```
Comment (derek-schultz) [inline docs_requirements.txt:2]: Is this intentional?

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    .pipe(uglify())
+    .pipe(rename({
+      suffix: '.min'
+    }))
+    .pipe(sourcemaps.write())
+    .pipe(gulp.dest(buildDestination));
+});
+
+gulp.task('watch', function() {
+  gulp.watch(files.scss, ['scss']);
```
Comment (derek-schultz) [inline gulpfile.js]: We need the same watch stuff here

File: orchestra/templates/orchestra/base.html [hunk 4]
```diff
@@ -78,13 +75,11 @@
     {% endblock %}
 
     <!-- js placed at the end of the document so the pages load faster -->
-    {% compress js %}
-    <script src="{% static "orchestra/common/js/lib/jquery.js" %}"></script>
+    <script src="{% static "dist/orchestra/common/js/lib/jquery.min.js" %}"></script>
```
Comment (derek-schultz) [inline orchestra/templates/orchestra/base.html:82]: Should we be minifying jquery ourselves?

File: docs_requirements.txt (hunk 2 above)
Comment (joshblum) [inline docs_requirements.txt:2]: yeah we forgot it in the 1.9 upgrade pass

File: orchestra/templates/orchestra/base.html (hunk 4 above)
Comment (derek-schultz) [inline orchestra/templates/orchestra/base.html:82]: It just seems weird

Comment (joshblum) [inline orchestra/templates/orchestra/base.html:82]: i don't think we should include any libs inside of the package, but we should move that do a different PR

File: docs/source/contributing.rst (hunk 1 above)
Comment (joshblum) [inline docs/source/contributing.rst]: done

File: gulpfile.js (hunk 3 above)
Comment (joshblum) [inline gulpfile.js]: done
